engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -103,11 +104,8 @@ public AnnotationMetaDataProvider(ConstraintHelper constraintHelper,
 		this.typeResolutionHelper = typeResolutionHelper;
 		this.valueExtractorManager = valueExtractorManager;
 		this.annotationProcessingOptions = annotationProcessingOptions;
-		this.configuredBeans = new ConcurrentReferenceHashMap<>(
-				DEFAULT_INITIAL_CAPACITY,
-				SOFT,
-				SOFT
-		);
+
+		this.objectBeanConfigurationReference = new AtomicReference<>( new SoftReference<>( retrieveBeanConfiguration( Object.class ) ) );
```

Comment:
So the object metadata is now always initialized. We can move it to a DCL but I thought with the SoftReference, it would be a bit complicated to follow. Opinions welcome.